waiter/test/waiter/test_helpers.clj

```diff
@@ -93,7 +163,33 @@
       (swap! running-tests #(dissoc % test-name))
       (with-test-out
         (println \tab (blue "FINISH:") test-name (cyan (format-duration elapsed-millis))
-                 (assoc @*report-counters* :running (count @running-tests))))
+                 (assoc @*report-counters* :running (count @running-tests)))
+        (when test-metrics-url
+          (let [{:keys [full-name name namespace]} (test-name-info (:var m))
+                num-fails (or (@test-name->num-fails-atom full-name) 0)
+                num-errors (or (@test-name->num-errors-atom full-name) 0)
+                num-passes (or (@test-name->num-passes-atom full-name) 0)
+                test-skipped? (= 0 (+ num-fails num-errors num-passes))
+                test-failed? (> (+ num-fails num-errors) 0)
+                result (if test-failed? "failed" (if test-skipped? "skipped" "passed"))
+                es-index (str "waiter-tests-" (du/date-to-str (t/now) (f/formatters :basic-date)))]
+            ;TODO: can check for outstanding commits: (println (jgit/git-status git-repo))
+            (post-json (str test-metrics-url "/" es-index "/test-result")
+                       (json/write-str {:timestamp (du/date-to-str (t/now))
```

Comment:
Should this map be generated by `cond->` and `assoc` as some of the fields may be `nil` when not configured?

Reply:
nil is fine